commit 5a493b9d0cb9bd16579e8f86036f7238476e0fd8
Author: Vlad Yasevich <vladislav.yasevich@hp.com>
Date:   Wed Sep 15 10:00:26 2010 -0400

    [Backported to Debian's 2.6.26 by dann frazier <dannf@debian.org>]
    
    sctp: Do not reset the packet during sctp_packet_config().
    
    sctp_packet_config() is called when getting the packet ready
    for appending of chunks.  The function should not touch the
    current state, since it's possible to ping-pong between two
    transports when sending, and that can result packet corruption
    followed by skb overlfow crash.
    
    Reported-by: Thomas Dreibholz <dreibh@iem.uni-due.de>
    Signed-off-by: Vlad Yasevich <vladislav.yasevich@hp.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/sctp/output.c b/net/sctp/output.c
index 6d45bae..9b38671 100644
--- a/net/sctp/output.c
+++ b/net/sctp/output.c
@@ -78,12 +78,6 @@ struct sctp_packet *sctp_packet_config(struct sctp_packet *packet,
 			  packet, vtag);
 
 	packet->vtag = vtag;
-	packet->has_cookie_echo = 0;
-	packet->has_sack = 0;
-	packet->has_auth = 0;
-	packet->has_data = 0;
-	packet->ipfragok = 0;
-	packet->auth = NULL;
 
 	if (ecn_capable && sctp_packet_empty(packet)) {
 		chunk = sctp_get_ecne_prepend(packet->transport->asoc);
